### PR TITLE
Exception-free wrapper for Laravel's `report` method.

### PR DESCRIPTION
**Pull request by nagmat84**
While I have been working on my streaming branch, I was bugged out by the stupidity of the Laravel `report` function. It is my ultimate goal to get my Linter down to zero findings in the code. But this requires just another wrapper for another Laravel method (`report` in this case). In order to keep it out of the other PRs, I created this PR.

See https://github.com/LycheeOrg/Lychee/pull/1315/commits/648981f7fb5dcea5c9454770893d7fe12849b75f for background information first.

### CODE REVIEW

**Review by qwerty287 — APPROVED** [review]
LGTM from code review, but untested.

File: app/Exceptions/Handler.php [hunk 1]
```diff
@@ -357,4 +357,39 @@ private function findCause(\Throwable $e): array
...
+	 * Normally, if one is inside a `catch`-block handling exceptions, one
+	 * does not like to deal with another (new) exception.
+	 * If `report` threw an exception, what should we do about it anyway?
+	 * Report it? ;-)
+	 * Even though the Laravel framework is very reluctant to document the
+	 * exceptions thrown by their methods, one of the few Laravel methods
+	 * which documents an exception surprisingly is
+	 * {@link \Illuminate\Contracts\Debug\ExceptionHandler::report()}.
+	 * Unfortunately, it is an unspecific `\Throwable`.
+	 * Even worse, we now that our own implementation of that method
```
Comment (kamil4) [inline app/Exceptions/Handler.php]:
```suggestion
	 * Even worse, we know that our own implementation of that method
```

File: app/Models/Extensions/ConfigsHas.php [hunk 2]
```diff
@@ -27,31 +29,26 @@ public static function hasExiftool(): bool
...
-					self::set('has_exiftool', 1);
-					$has_exiftool = true;
-				}
-			} catch (Exception $e) {
-				self::set('has_exiftool', 0);
-				$has_exiftool = false;
-				Logs::warning(__METHOD__, __LINE__, 'exec is disabled, has_exiftool has been set to 0.');
+			} catch (\Exception $e) {
+				$path = '';
+				Handler::reportSafely(new ExternalComponentMissingException('could not find exiftool, `has_exiftool` will be set to 0', $e));
```
Comment (kamil4) [inline app/Models/Extensions/ConfigsHas.php]: Nitpicking mode **on** :wink:

```suggestion
				Handler::reportSafely(new ExternalComponentMissingException('could not find exiftool; `has_exiftool` will be set to 0', $e));
```

File: app/Models/Extensions/ConfigsHas.php [hunk 3]
```diff
@@ -64,30 +61,25 @@ public static function hasFFmpeg(): bool
...
-					self::set('has_ffmpeg', 1);
-					$has_ffmpeg = true;
-				}
-			} catch (Exception $e) {
-				self::set('has_ffmpeg', 0);
-				$has_ffmpeg = false;
-				Logs::warning(__METHOD__, __LINE__, 'exec is disabled, set_ffmpeg has been set to 0.');
+			} catch (\Exception $e) {
+				$path = '';
+				Handler::reportSafely(new ExternalComponentMissingException('could not find ffmpeg, `has_ffmpeg` will be set to 0', $e));
```
Comment (kamil4) [inline app/Models/Extensions/ConfigsHas.php]:
```suggestion
				Handler::reportSafely(new ExternalComponentMissingException('could not find ffmpeg; `has_ffmpeg` will be set to 0', $e));
```

**Review by kamil4 — APPROVED** [review]
Untested but looks sane. I just added some comments about... comments.

**Pull request merged**